controllers/policyset/policyset_controller.go

```diff
@@ -159,6 +146,46 @@ func processPolicySet(ctx context.Context, c client.Client, plcSet *policyv1.Pol
 					placementsByBinding[placement.PlacementBinding] = plcPlacementToSetPlacement(*placement)
 				}
 			}
+
+			// create list of all relevant clusters
+			clusters := []string{}
+			for pbName := range placementsByBinding {
```

Comment:
You already have the placement info in `placementsByBinding` map. Isn't it easier to query the placementDecision and placementrule to get a list of applicable clusters and then filter out the policy.status to generate the results?

Reply:
`GetPlacementDecisions` function generates `placements` which you don't really need here and you have to filter them again.

Reply:
I thought placementDecision didn't show up in policy status?

Reply:
You need to query `placementDecision`and `placementrule` in order know which cluster needs to apply. That's what `GetPlacementDecisions` does. However, this function also process and return a list of placements which you don't need and you didn't use it. It's better to refactor the function to return `placements` and `decisions` separately.